core/include/kernel/interrupt.h

```diff
@@ -14,11 +15,23 @@
 
 struct itr_chip {
 	const struct itr_ops *ops;
-	int (*dt_get_irq)(const uint32_t *properties, int len);
+	/*
+	 * dt_get_irq - parse a device tree interrupt property
+	 *
+	 * @properties raw interrupt property from device tree
+	 * @len number of properties
+	 * @type irq type (IRQ_TYPE_* defines) if specified by interrupt
+	 * property or IRQ_TYPE_NONE if not. Pointer is never NULL.
+	 * @prio irq priority if specified by interrupt property. Pointer is
+	 * never NULL.
```

Comment:
```
	 * @prio Output interrupt priority or 0.
```
I think the callback should handle NULL `type` and `prio` pointers.
(` * @prio Output interrupt priority value (0 if unknown) if not NULL.`)

Reply:
Ok, I wanted to do this in the upper layer to avoid having each implementation to do it but if you are ok with that, the code that uses intermediate local variable can even be removed. I will modify that

Reply:
Ok I removed this and allows passing NULL pointers.